### PR TITLE
updated src and UT files for implementation of "selectively ignore some measure"

### PR DESCRIPTION
**Pull request by Madhu-NEC**
PR is with correspond to [iotagent-json#416](https://github.com/telefonicaid/iotagent-json/issues/416).

### CODE REVIEW

File: lib/commonConfig.js [hunk 1]
```diff
@@ -235,6 +236,11 @@ function processEnvironmentVariables() {
         config.defaultResource = process.env.IOTA_DEFAULT_RESOURCE;
     }
 
+    // Default explicitAttrs
+    if (process.env.IOTA_DEFAULT_EXPLICITATTRS !== undefined) {
```
Comment (fgalan) [inline lib/commonConfig.js]: A couple of comments:

1. The flag (and associated env var) should be documented in installagionguide.md. Have a look to `appendMode` and `timestamp` documented there and do it in a similar way. Don't forget to include the last sentece: "This flag is overwritten by ... flag in group or device provision."
2. Change the name of the configuration parameter `defaultExplicitAttrs` -> `explicitAttrs` (consecuently, using IOTA_EXPLICT_ATTRS as env var). This way, the name of the configuraiton parameter will match with the one used in the JSON payload at group or device provisining.

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -1,2 +1,3 @@
+Added feature: selectively ignore some of the measures (iotagent-json#416)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]:
```suggestion
Add: explicitAttrs flag (in configuration and also group/device provisioning) to progress only the measures correspoing to attributes declared in attributes array (iotagent-json#416)
```

I think that way the description is more precise. What do you think?

File: lib/services/devices/deviceRegistryMongoDB.js [hunk 3]
```diff
@@ -61,7 +61,7 @@ function storeDevice(newDevice, callback) {
     var deviceObj = new Device.model(),
         attributeList = ['id', 'type', 'name', 'service', 'subservice', 'lazy', 'commands', 'staticAttributes',
             'active', 'registrationId', 'internalId', 'internalAttributes', 'resource', 'apikey', 'protocol',
-            'endpoint', 'transport', 'polling', 'timestamp', 'autoprovision'];
+            'endpoint', 'transport', 'polling', 'timestamp', 'autoprovision','explicitAttrs'];
```
Comment (fgalan) [inline lib/services/devices/deviceRegistryMongoDB.js]:
```suggestion
            'endpoint', 'transport', 'polling', 'timestamp', 'autoprovision', 'explicitAttrs'];
```

Just a small coding style fix ;)

File: test/unit/examples/deviceProvisioningRequests/provisionExplicitAttrs.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (fgalan) [inline test/unit/examples/deviceProvisioningRequests/provisionExplicitAttrs.json]: What about group provisioning with/without explicitAttrs? Is the any test covering that case?

File: CHANGES_NEXT_RELEASE (hunk 2 above)
Comment (Madhu-NEC) [reply CHANGES_NEXT_RELEASE]: Fixed in 339d67a

File: lib/services/devices/deviceRegistryMongoDB.js (hunk 3 above)
Comment (Madhu-NEC) [reply lib/services/devices/deviceRegistryMongoDB.js]: Fixed in 051c7d9

File: lib/commonConfig.js (hunk 1 above)
Comment (Madhu-NEC) [reply lib/commonConfig.js]: > 1. The flag (and associated env var) should be documented in installagionguide.md. Have a look to appendMode and timestamp documented there and do it in a similar way. Don't forget to include the last sentece: "This flag is overwritten by ... flag in group or device provision."

Fixed in ac1120b

> 2. Change the name of the configuration parameter defaultExplicitAttrs -> explicitAttrs (consecuently, using IOTA_EXPLICT_ATTRS as env var). This way, the name of the configuraiton parameter will match with the one used in the JSON payload at group or device provisining.

Fixed in 051c7d9

File: test/unit/examples/deviceProvisioningRequests/provisionExplicitAttrs.json (hunk 4 above)
Comment (Madhu-NEC) [reply test/unit/examples/deviceProvisioningRequests/provisionExplicitAttrs.json]: Group provisioning test cases are covered in **test/unit/provisioning/device-group-api-test.js** file.

Comment (fgalan) [reply test/unit/examples/deviceProvisioningRequests/provisionExplicitAttrs.json]: NTC

**Comment by Madhu-NEC** [comment]
Hi @fgalan ,
I have resolved the conflict as well as updated the PR, request you to please merge the PR.

**Comment by fgalan** [comment]
With the creation of PR https://github.com/telefonicaid/iotagent-ul/pull/430 (which I understand completes the selectively ignore some measures functionality along all the main agents) I think is the moment of updating this PR with master and solve the current conflict.

@Madhu-NEC could you manage? Thanks!

**Comment by fgalan** [comment]
Sorry... I didn't realize about new PR https://github.com/telefonicaid/iotagent-node-lib/pull/873

I undesrtand that #873 is an updated version of this one. Should PR #835 be closed?

**Comment by Madhu-NEC** [comment]
> I undesrtand that #873 is an updated version of this one. Should PR #835 be closed?

Hi @fgalan ,
As #873 is updated version of this one. We can close this PR #835 .
Thanks!

**Pull request closed without merging**